velox/exec/ExchangeSource.h

```diff
@@ -86,8 +86,14 @@ class ExchangeSource : public std::enable_shared_from_this<ExchangeSource> {
   /// backward compatibility (e.g. communicating with coordinator), we allow
   /// small data (1MB) to be returned.
   virtual folly::SemiFuture<Response> requestDataSizes(
-      uint32_t /*maxWaitSeconds*/) {
-    VELOX_NYI();
+      uint32_t maxWaitSeconds) = 0;
+
+  const std::vector<int64_t>& remainingBytes() const {
```

Comment:
Should the caller hold a queue lock when making this call? If so, let's rename to remainingBytesLocked.